packages/react-query/src/__tests__/suspense.test.tsx

```diff
@@ -887,4 +887,52 @@ describe('useSuspenseQueries', () => {
     await waitFor(() => rendered.getByText('data: 1,2'))
     expect(refs[0]).toBe(refs[1])
   })
+
+  // this addresses the following issue:
+  // https://github.com/TanStack/query/issues/6344
+  it('should suspend on offline when query changes, and data should not be undefined', async () => {
+    function Page() {
+      const [id, setId] = React.useState(0)
+
+      const { data } = useSuspenseQuery({
+        queryKey: [id],
+        queryFn: () => Promise.resolve(`Data ${id}`),
+      })
+
+      // defensive guard here
+      if (data === undefined) {
```

Comment:
Commenting the resolution in `suspense.ts` will make lead to undefined and thus throwing this error